Content.Server/Weapons/Melee/EnergyShieldSystem.cs

```diff
@@ -0,0 +1,114 @@
+using Content.Server.CombatMode.Disarm;
+using Content.Server.Kitchen.Components;
+using Content.Shared.Interaction;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Item;
+using Content.Shared.Temperature;
+using Content.Shared.Toggleable;
+using Content.Shared.Tools.Components;
+using Robust.Shared.Player;
+
+namespace Content.Shared.Weapons.Melee.EnergyShield;
+
+public sealed class EnergyShieldSystem : EntitySystem
+{
+    [Dependency] private readonly SharedItemSystem _item = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedAppearanceSystem _appearance = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<ItemToggleComponent, UseInHandEvent>(OnUseInHand);
+        SubscribeLocalEvent<ItemToggleComponent, InteractUsingEvent>(OnInteractUsing);
+        SubscribeLocalEvent<ItemToggleComponent, IsHotEvent>(OnIsHotEvent);
+        SubscribeLocalEvent<ItemToggleComponent, EnergyShieldDeactivatedEvent>(TurnOff);
+        SubscribeLocalEvent<ItemToggleComponent, EnergyShieldActivatedEvent>(TurnOn);
+    }
```

Comment:
So what I meant by this was that you should make a separate Component and System from Energy Shield and Energy Sword as ItemToggleComponent/ItemToggleComponentSystem. Then you add the ItemToggleComponent to the Energy Shield yml with the desired values. If you need anything else, then you can make an Energy Shield Component/System as needed.

Reply:
That is, to edit and Energy Sword?

Reply:
No need to touch energy sword on this PR, that can be done in a separate PR

Reply:
Then it's done.